zebra-network/src/address_book.rs

```diff
@@ -266,6 +266,18 @@ impl AddressBook {
         self.local_listener
     }
 
+    /// Get the active addresses in `self` in random order with sanitized timestamps,
+    /// including our local listener address.
+    pub fn sanitized_window(&self) -> Vec<MetaAddr> {
+        let now = Utc::now();
+        let mut peers = self.sanitized(now);
+        let address_limit = peers.len().div_ceil(ADDR_RESPONSE_LIMIT_DENOMINATOR);
+        let address_limit = MAX_ADDRS_IN_MESSAGE.min(address_limit);
+        peers.truncate(address_limit);
+
+        peers
+    }
+
     /// Get the active addresses in `self` in random order with sanitized timestamps,
     /// including our local listener address.
     pub fn sanitized(&self, now: chrono::DateTime<Utc>) -> Vec<MetaAddr> {
```

Comment:
> Let's make the less secure one private

It would require some test changes and there are other methods that give out the same information, could we create an issue for it?

Renamed to `fresh_get_addr_response()`.

Reply:
See commit [87a226f](https://github.com/ZcashFoundation/zebra/pull/7955/commits/87a226f3e4b898f89258eca42ec60902fce58bfc) for this change.

If we've changed the method used by GetAddr, do those tests also need to use the new method?

I think this test does:
https://github.com/ZcashFoundation/zebra/blob/c3b11e325bccd24c16a993d46db7f847303c970b/zebra-network/src/address_book/tests/prop.rs#L39

But the other test that uses `sanitized()` doesn't. And maybe we need another test for this method specifically?

Reply:
> If we've changed the method used by GetAddr, do those tests also need to use the new method?

I think so, some of the new method wasn't covered. Fixed in https://github.com/ZcashFoundation/zebra/pull/7955/commits/2ee9a207342a94aed5d3e871c1304673b4839e00.